src/views/Common/TextInput/TextInput.tsx

```diff
@@ -11,6 +11,7 @@ interface IProps {
     labelStyle?: React.CSSProperties;
     barStyle?: React.CSSProperties;
     value?: string;
+    handlerForEnterKey?: () => void;
```

Comment:
To make this more general, please rename this props `onKeyUp?: (event: React.KeyboardEvent<HTMLInputElement>) => void;`

Instead of passing only the `Enter` handler alone, pass whole:

```
const handleKeyUp = (event: React.KeyboardEvent<HTMLInputElement>) => {
        if (handlerForEnterKey) {
            if (event.key === 'Enter') {
                handlerForEnterKey();
            }
        }
    }
```

As props value.

Than `handleKeyUp` should be defined in `InsertLabelNamesPopup.tsx`. Than potentially in the future if you would like to do some other action on other keys you would be able to pass it via this single callback.